source/dofs/dof_handler_policy.cc

```diff
@@ -1209,126 +1843,57 @@ namespace internal
           const_cast<dealii::Triangulation<dim, spacedim> &>(
             dof_handler.get_triangulation())
             .load_user_flags_quad(user_flags);
-
-          return dof_identities;
         }
 
 
 
         /**
-         * Once degrees of freedom have been distributed on all cells, see if
-         * we can identify DoFs on neighboring cells. This function does nothing
-         * on regular DoFHandlers, but goes through vertices, lines, and quads
-         * for hp::DoFHandler objects.
+         * After DoF unification in Phase 2, we may still have invalid DoF
+         * indices left on ghost interfaces that are dominated by a
+         * FiniteElement object of a neighboring cell. After the first ghost
```

Comment:
It could actually be either a locally owned or ghost cell. The crucial point is that these remaining invalid dofs are dominated by something that we did not know before.

I'll extend the comment on that.